src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -45,5 +47,1075 @@ class Outer
+        }
+
+        [Fact]
+        public void SignatureUsage_06()
+        {
+            var source = """
+                file class C
+                {
+                }
+
+                delegate void Del1(C c); // 1
+                delegate C Del2(); // 2
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (5,15): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'Del1'.
+                // delegate void Del1(C c); // 1
+                Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "Del1").WithArguments("C", "Del1").WithLocation(5, 15),
+                // (6,12): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'Del2'.
+                // delegate C Del2(); // 2
+                Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "Del2").WithArguments("C", "Del2").WithLocation(6, 12));
+        }
+
+        [Fact]
+        public void SignatureUsage_07()
+        {
+            var source = """
+                file class C
+                {
+                }
+
+                class D
+                {
+                    public static D operator +(D d, C c) => d; // 1
+                    public static C operator -(D d1, D d2) => new C(); // 2
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (7,30): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+                //     public static D operator +(D d, C c) => d; // 1
+                Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "+").WithArguments("C", "D").WithLocation(7, 30),
+                // (8,30): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+                //     public static C operator -(D d1, D d2) => new C(); // 2
+                Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "-").WithArguments("C", "D").WithLocation(8, 30));
+        }
+
+        [Fact]
+        public void SignatureUsage_08()
+        {
+            var source = """
+                file class C
+                {
+                }
+
+                class D
+                {
+                    public D(C c) { } // 1
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (7,12): error CS9300: File type 'C' cannot be used in a member signature in non-file type 'D'.
+                //     public D(C c) { } // 1
+                Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "D").WithArguments("C", "D").WithLocation(7, 12));
+        }
+
+        [Fact]
+        public void SignatureUsage_09()
```

Comment:
Not blocking this PR, but it would be good to get a broader coverage of usages, including positive ones: locals, constraints (methods, types, delegates), base type (`file class Outer { class Inner : Outer { } }` and `file class Outer { class Inner : Base<Outer> { } }`) , interface, delegate types, lambdas, local functions, attributes, ...